model/image.go

```diff
@@ -69,14 +70,158 @@ func (img *Image) GetSamples() []uint32 {
 // SetSamples convert samples to byte-data and sets for the image.
 func (img *Image) SetSamples(samples []uint32) {
 	resampled := sampling.ResampleUint32(samples, int(img.BitsPerComponent), 8)
```

Comment:
The `SetSamples` method is quite expensive memory-wise and should be avoided when possible. It's better to avoid it in the conversion methods of the most common colorspaces and work directly on the byte data. The same goes for `GetSamples`.

Reply:
Maybe worth adding a warning message advising not to use it?  (Similar to your comment above).